### ISSUE TITLE
Information about the pt_BR language

### ISSUE THREAD

**Issue by leofontenelle** [translate/translate#630]
This information is based on what I saw in lang/common.py. Please let me know if you need further information.

code = "pt_BR"
fullname = "Portuguese (Brazil)"
nplurals = "2"
pluralequation = "n>1"
validaccel = u"12345667890ABCDEFGHIJKLMNOPQRSTUVXYZabcdefghijklmnopqrstuvxyz"

The preferred quoting style is either "double straight quotes" or “double curly quotes”; 'single straight quotes' or their curly equivalents happen inside the double ones. Example: "The dialog had an 'OK' button." The other quotes are either obsolete or plain wrong.

We don't use inverted punctuation or other specific ones, but most misc. punctuation can happen in pt_BR. "·" (middle dot) is not valid, and marking it as wrong might help the Brazilian OpenOffice.org and GNOME translation teams fix some translations.

I don't think any pofilter test is completely invalid in pt_BR, except maybe for doublespacing. It can be removed if other test verifies the multiple spaces used by developers for padding in the user interface. I don't believe there should be any language-specific test.

About punctuation/spacing, we use "[:,;?!] ", there's no double spacing. There's no start punctuation (just like English).

**Comment by unho** [comment]
> This information is based on what I saw in `lang/common.py`. Please let me know if you need further information.

> ```python
code = "pt_BR"
fullname = "Portuguese (Brazil)"
nplurals = "2"
pluralequation = "n>1"
validaccel = u"12345667890ABCDEFGHIJKLMNOPQRSTUVXYZabcdefghijklmnopqrstuvxyz"
```

Just added the missing `validaccel`.

> The preferred quoting style is either “double straight quotes” or “double curly quotes”; ‘single straight quotes’ or their curly equivalents happen inside the double ones. Example: “The dialog had an ‘OK’ button.” The other quotes are either obsolete or plain wrong.

Adding a specific check for this might produce too much false positives.

Replacing the quotes appearances might produce unwanted errors which should be checked with some Brazilian translator. Also the lack of preference between straight and curly double quotes complicates this more (French has a replacing code to `«»`).

> We don’t use inverted punctuation or other specific ones, but most misc. punctuation can happen in pt_BR. “·” (middle dot) is not valid, and marking it as wrong might help the Brazilian OpenOffice.org and GNOME translation teams fix some translations.

Excluded middle dot from punctuation.

> I don’t think any pofilter test is completely invalid in pt_BR, except maybe for doublespacing. It can be removed if other test verifies the multiple spaces used by developers for padding in the user interface. I don’t believe there should be any language-specific test.

We keep the default tests.

> About punctuation/spacing, we use `"[:,;?!] "`, there’s no double spacing. There’s no start punctuation (just like English).

We keep the defaults.

**Comment by josephgbr** [comment]
> > The preferred quoting style is either “double straight quotes” or “double curly quotes”; ‘single straight quotes’ or their curly equivalents happen inside the double ones. Example: “The dialog had an ‘OK’ button.” The other quotes are either obsolete or plain wrong.
> Adding a specific check for this might produce too much false positives.

> Replacing  the quotes appearances might produce unwanted errors which should be checked with some Brazilian translator.

@unho Hello, I'm a Brazilian Portuguese Translator. May I help somehow with testing such feature? While I may not be able to provide patches, I surely can test them.

**Comment by unho** [comment]
@josephgbr We can arrange some script to automatically check several thousands of strings if you have access to that amount of strings. Of course before that we have to write that new checker code. Given our current time constraints that won't happen on our side until after next TTK and Pootle releases.

**Comment by josephgbr** [comment]
@unho I could use all pt_BR GNOME's and Translation Project's catalog files - is it enough? Just let em know and provide instructions, when available.

**Comment by unho** [comment]
@josephgbr Yes, that should work, at least to tell if something goes obviously wrong.
